docs/machine-learning/tutorials/sentiment-analysis.md

```diff
@@ -134,82 +140,88 @@ Remove the existing class definition and add the following code, which has two c
 
-You add `async` to `Main` with a <xref:System.Threading.Tasks.Task> return type because you're saving the model to a zip file later, and the program needs to wait until that external task completes.
+[!code-csharp[CreateMLContext](../../../samples/machine-learning/tutorials/SentimentAnalysis/Program.cs#3 "Create the ML Context")]
 
-> [!NOTE]
-> An *async main* method enables you to use `await` in your `Main` method. For more information, see the
-[async main](../../../docs/csharp/programming-guide/main-and-command-args/index.md) topic in the C# programming guide.
+Next, to setup for data loading initialize the `_textLoader` global variable in order to reuse it.  Notice that we are using a <xref:Microsoft.ML.Runtime.DataLoadSaveOperations.TextReader>. When you create a `TextLoader` using a `TextReader`, you pass in the context needed and the <xref:Microsoft.ML.Runtime.Data.TextLoader.Arguments> class which enables customization.
+
+ Specify the data schema by passing an array of <xref:Microsoft.ML.Runtime.Data.TextLoader.Column> objects to the loader containing all the column names and their types. We defined the data schema previously when we created our `SentimentData` class. For our schema, the first column (Label) is a Boolean (the prediction) and the second column (SentimentText) is the feature of type text/string used for predicting the sentiment.
+The `TextReader` class returns a fully initialized <xref:Microsoft.ML.Runtime.Data.TextLoader>  
+
+To initialize the `_textLoader` global variable in order to reuse it for the needed datasets, add the following code after the  `mlContext` initialization:
 
-Replace the `Console.WriteLine("Hello World!")` line with the following code in the `Main` method:
+[!code-csharp[initTextReader](../../../samples/machine-learning/tutorials/SentimentAnalysis/Program.cs#4 "Initialize the TextReader")]
 
-[!code-csharp[Train](../../../samples/machine-learning/tutorials/SentimentAnalysis/Program.cs#3 "Train your model")]
+Add the following as the next line of code in the `Main` method:
+
+[!code-csharp[Train](../../../samples/machine-learning/tutorials/SentimentAnalysis/Program.cs#5 "Train your model")]
 
 The `Train` method executes the following tasks:
 
-* Loads or ingests the data.
-* Preprocesses and featurizes the data.
+* Loads the data.
+* Extracts and transforms the data.
 * Trains the model.
 * Predicts sentiment based on test data.
+* Returns the model.
 
 Create the `Train` method, just after the `Main` method, using the following code:
 
 ```csharp
-public static async Task<PredictionModel<SentimentData, SentimentPrediction>> Train()
+ public static ITransformer Train(MLContext mlContext, string dataPath)
 {
 
 }
 ```
 
-## Ingest the data
+Notice that two parameters are passed into the Train method; a MLContext for the context (`mlContext`), and a string for the dataset path (`dataPath`). We're going to use this method more than once for training and testing.
```

Comment:
`MLContext` should be an xref.

Reply:
Even if I previously cross referenced it in line 147?

Reply:
I missed that. Code fence is fine then. That way, it doesn't get translated.